### PR TITLE
feat(components): add dialog component

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
## What/Why?
This PR adds Dialog component & updates Storybook.

## Testing
Locally

https://github.com/bigcommerce/catalyst/assets/67792608/5be838b0-fd70-49df-a591-2e5b4c3181e4

### CODE REVIEW

File: packages/components/src/components/dialog/dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import { cn } from '~/lib/utils';
+
+const Dialog = DialogPrimitive.Root;
+
+const DialogTrigger = forwardRef<
+  ElementRef<typeof DialogPrimitive.Trigger>,
+  ComponentPropsWithoutRef<typeof DialogPrimitive.Trigger>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Trigger className={cn('', className)} ref={ref} {...props} />
```
Comment (jorgemoya) [inline packages/components/src/components/dialog/dialog.tsx:12]:
```suggestion
  <DialogPrimitive.Trigger className={cn(className)} ref={ref} {...props} />
```

File: packages/components/src/components/dialog/dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+));
+
+DialogTrigger.displayName = DialogPrimitive.Trigger.displayName;
+
+const DialogPortal = forwardRef<
+  ElementRef<'div'>,
```
Comment (jorgemoya) [inline packages/components/src/components/dialog/dialog.tsx:18]:
```suggestion
  ElementRef<typeof DialogPrimitive.Portal>,
```

File: packages/components/src/components/dialog/dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  <DialogPrimitive.Description className={cn('', className)} ref={ref} {...props} />
+));
+
+DialogDescription.displayName = DialogPrimitive.Description.displayName;
+
+const DialogCancel = forwardRef<
+  ElementRef<typeof DialogPrimitive.Cancel>,
+  ComponentPropsWithoutRef<typeof DialogPrimitive.Cancel>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Cancel className={cn('', className)} ref={ref} {...props} />
```
Comment (jorgemoya) [inline packages/components/src/components/dialog/dialog.tsx:83]:
```suggestion
  <DialogPrimitive.Cancel className={cn(className)} ref={ref} {...props} />
```

File: packages/components/src/components/dialog/dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+));
+
+DialogCancel.displayName = DialogPrimitive.Cancel.displayName;
+
+const DialogAction = forwardRef<
+  ElementRef<typeof DialogPrimitive.Action>,
+  ComponentPropsWithoutRef<typeof DialogPrimitive.Action>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Action className={cn('', className)} ref={ref} {...props} />
```
Comment (jorgemoya) [inline packages/components/src/components/dialog/dialog.tsx:92]:
```suggestion
  <DialogPrimitive.Action className={cn(className)} ref={ref} {...props} />
```

**Review by jorgemoya — COMMENTED** [review]
Can you provide screenshots of this new component? There also seems to be an issue with your dependencies.

File: packages/components/src/components/dialog/dialog.tsx (hunk 1 above)
Comment (bc-alexsaiannyi) [reply packages/components/src/components/dialog/dialog.tsx:12]: I aimed to check if I need extra styles in a few places and forgot to clean it up. Thanks for catching this, Jorge🙏

File: packages/components/src/components/dialog/dialog.tsx (hunk 2 above)
Comment (bc-alexsaiannyi) [reply packages/components/src/components/dialog/dialog.tsx:18]: when I tried to provide `ref` to `Portal` I got an error:
Screenshot 2024-04-16 at 09 43 07

So I wrapped `Portal` into `<div/>` and put `ref` on `div`...

Comment (jorgemoya) [reply packages/components/src/components/dialog/dialog.tsx:18]: Ahh, got it!

**Review by jorgemoya — APPROVED** [review]

File: apps/docs/stories/dialog.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          <div className="flex flex-col gap-2 p-6 lg:flex-row">
+            <DialogAction asChild>
+              <Button className="w-full lg:w-fit" variant="primary">
+                Yes
+              </Button>
+            </DialogAction>
+            <DialogCancel asChild>
+              <Button className="w-full lg:w-fit" variant="subtle">
+                Cancel
+              </Button>
```
Comment (yurytut1993) [inline apps/docs/stories/dialog.stories.tsx:51]: should this text be hardcoded?

**Review by yurytut1993 — APPROVED** [review]
👌

**Pull request merged**